src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -303,9 +290,17 @@ class UserService(
     )
   }
 
-  private fun findProbationRegionFromArea(probationArea: StaffProbationArea): ProbationRegionEntity? {
-    return probationAreaProbationRegionMappingRepository
+  private fun findProbationRegionFromArea(deliusUser: StaffUserDetails): ProbationRegionEntity {
+    val probationArea = deliusUser.probationArea
+    val probationRegion = probationAreaProbationRegionMappingRepository
       .findByProbationAreaDeliusCode(probationArea.code)?.probationRegion
+
+    return probationRegion ?: if (assignDefaultRegionToUsersWithUnknownRegion) {
+      log.warn("Unknown probation region code '${probationArea.code}' for user '${deliusUser.username}', assigning a default region of 'North West'.")
+      return probationRegionRepository.findByName("North West")!!
+    } else {
+      throw BadRequestProblem(errorDetail = "Unknown probation region code '${probationArea.code}' for user '${deliusUser.username}'")
```

Comment:
we now throw this on update of existing users, previously we would just ignore an unknown probation region and leave as-is. Ed, do you think there's any change that this'll cause an issue for CAS2/3?